src/main/java/com/almightyqa/TestUtils.java

```diff
@@ -0,0 +1,122 @@
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.stream.IntStream;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class TestUtils {
+
+    public static final Random RANDOM = new Random();
+    public static final Faker FAKER = new Faker();
+
+    public static final Gson GSON = new Gson();
+
+    public static int getRandomInt(int... bound) {
+        return bound.length == 0 ? RANDOM.nextInt() : RANDOM.nextInt(bound[0]);
+    }
+
+    public static long getRandomLong(long... bound) {
+        return bound.length == 0 ? RANDOM.nextInt() : RANDOM.nextLong(bound[0]);
+    }
+
+    public static long getRandomNumberOfGivenSize(int numberOfDigits) {
+        long min = 1;
+        if(numberOfDigits>19){
+            throw new IllegalArgumentException("Max number for long is 19 digits long");
+        }
+        for (int i = 1; i < numberOfDigits; i++) {
+            min *= 10;
+        }
+        return min + getRandomLong(9L * min) - 1;
+    }
+
+    public static String getRandomEmail(final String domain) {
+        return FAKER.name().firstName() +
+                Math.abs(getRandomInt()) +
+                "@" + domain + ".com";
+    }
+
+    public static String getRandomBankAccount() {
+        return String.format("%08d", getRandomInt(100000000));
+    }
+
+    public static String getRandomTaxpayerId() {
+        return 11 + getRandomBankAccount();
+    }
+
+    public static <T extends Enum> T getRandomEnum(Class<T> enumClass) {
+        final T[] enumConstants = enumClass.getEnumConstants();
+        return Arrays.asList(enumConstants).get(getRandomInt(enumConstants.length));
+    }
+
+    public static <T> T getOneOf(T... items) {
+        final int randomElementIndex = getRandomInt(items.length);
+        return items[randomElementIndex];
+    }
+
+
+    public static String formatDate(final LocalDate date) {
+        return DateTimeFormatter.ofPattern("yyyy-MM-dd").format(date);
+    }
+
+    public static Object[][] dataProviderFromList(final List<?> listOfItems) {
+        final Object[][] dataProvider = new Object[listOfItems.size()][];
+        for (int i = 0; i < listOfItems.size(); i++) {
+            dataProvider[i] = new Object[]{listOfItems.get(i)};
+        }
+        return dataProvider;
+    }
+
+    public static String getRandomPhoneNumber() {
+        return String.valueOf(getRandomNumberOfGivenSize(8));
+    }
+
+    public static String uppercaseLettersOnly(final String text) {
+        return text.replaceAll("[\\s|[^\\w\\s]]", "").toUpperCase();
+    }
+
+    public static String generateRandomLetter(final int amount) {
+        StringBuilder sb = new StringBuilder().append("TEST");
+        IntStream.range(0, amount - 4).forEach(i -> sb.append((char) (TestUtils.getRandomInt(26) + 'A')));
+        return sb.toString();
+    }
+
+    public static Object[][] createCombinationsOfLists(final List<?> items1, final List<?> items2) {
```

Comment:
Question - don't you need list of all combinations?
if so, you can use (try it before using)
    public static List<Object[]> createCombinationsOfLists(List<?> items1, List<?> items2) {
        final List<Object[]> combinations = new ArrayList<>();

        for (Object item1 : items1) {
            for (Object item2 : items2) {
                combinations.add(new Object[]{item1, item2});
            }
        }

        return combinations;
    }